Essentials/src/main/java/com/earth2me/essentials/utils/AdventureUtil.java

```diff
@@ -1,41 +1,73 @@
-    public static final char KEZZ_MAGIC_CHAR = 0x7f;
-    private static final Pattern NAMED_PATTERN = Pattern.compile(KEZZ_MAGIC_CHAR + "[0-9a-fk-orA-FK-OR]");
-    private static final Pattern HEX_PATTERN = Pattern.compile(KEZZ_MAGIC_CHAR + "x((?:" + KEZZ_MAGIC_CHAR + "[0-9a-fA-F]){6})");
+    private static final MiniMessage MINI_MESSAGE_INSTANCE;
+    private static IEssentials ess;
+    private static final Pattern NAMED_PATTERN = Pattern.compile(ChatColor.COLOR_CHAR + "[0-9a-fk-orA-FK-OR]");
+    private static final Pattern HEX_PATTERN = Pattern.compile(ChatColor.COLOR_CHAR + "x((?:" + ChatColor.COLOR_CHAR + "[0-9a-fA-F]){6})");
     private static final String LOOKUP = "0123456789abcdefklmnor";
     private static final String[] MINI_TAGS = new String[] {"black", "dark_blue", "dark_green", "dark_aqua", "dark_red", "dark_purple", "gold", "gray", "dark_gray", "blue", "green", "aqua", "red", "light_purple", "yellow", "white", "obf", "b", "st", "u", "i", "reset"};
+    private static final NamedTextColor[] COLORS = new NamedTextColor[] {NamedTextColor.BLACK, NamedTextColor.DARK_BLUE, NamedTextColor.DARK_GREEN, NamedTextColor.DARK_AQUA, NamedTextColor.DARK_RED, NamedTextColor.DARK_PURPLE, NamedTextColor.GOLD, NamedTextColor.GRAY, NamedTextColor.DARK_GRAY, NamedTextColor.BLUE, NamedTextColor.GREEN, NamedTextColor.AQUA, NamedTextColor.RED, NamedTextColor.LIGHT_PURPLE, NamedTextColor.YELLOW, NamedTextColor.WHITE};
 
     static {
         final LegacyComponentSerializer.Builder builder = LegacyComponentSerializer.builder().flattener(ComponentFlattener.basic()).useUnusualXRepeatedCharacterHexFormat();
         if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_16_1_R01)) {
             builder.hexColors();
         }
         LEGACY_SERIALIZER = builder.build();
+
+        MINI_MESSAGE_INSTANCE = MiniMessage.builder()
+                .tags(TagResolver.builder()
+                        .resolvers(TagResolver.standard())
+                        .resolver(TagResolver.resolver("primary", supplyTag(true)))
+                        .resolver(TagResolver.resolver("secondary", supplyTag(false)))
+                        .build())
+                .build();
+
     }
 
     private AdventureUtil() {
     }
 
-    public static Component toComponent(final String text) {
+    public static void setEss(final IEssentials ess) {
+        AdventureUtil.ess = ess;
+    }
+
+    public static MiniMessage miniMessage() {
+        return MINI_MESSAGE_INSTANCE;
+    }
+
+    public static Component deserializeLegacy(final String text) {
         return LEGACY_SERIALIZER.deserialize(text);
     }
 
+    public static String serializeLegacy(final Component component) {
+        return LEGACY_SERIALIZER.serialize(component);
+    }
+
+    public static String miniToLegacy(final String format) {
+        return serializeLegacy(miniMessage().deserialize(format));
+    }
+
     public static String minifyLegacy(String text) {
```

Comment:
Took a minute to figure out what these all do. Could maybe make the method names more consistent (at least change `minifyLegacy` to `legacyToMini`, not so sure about the `(de)serializeLegacy`)